### ISSUE TITLE
Adding a magnet with trailing newline returns a valid handle with all-zero info_hash

### ISSUE THREAD

**Issue by cas--** [arvidn/libtorrent#1716]
A code snippet for clarity:
```
import libtorrent as lt
sess = lt.session()
handle = sess.add_torrent(
    {'save_path':'', 'url':'magnet:?xt=urn:btih:5FKOMYDL555S6I6YNFHUSUKCSEWGOWC3\n'})
print(str(handle.info_hash()))
```
```
0000000000000000000000000000000000000000
```
It would be better to test if a magnet uri and raise an exception if invalid than return a valid handle.

All lt versions afaik...

**Comment by arvidn** [comment]
you mean "url" instead of "uri", right?

**Comment by cas--** [comment]
Ah yes! I actually noticed it first with deprecated `add_magnet_uri()`

**Comment by cas--** [comment]
Though the fact that `uri` also gives back a valid handle is another bug?

**Comment by aldenml** [comment]
did you ask for `handle.is_valid()`?

**Comment by cas--** [comment]
```
>>> import libtorrent as lt
... 
... sess = lt.session()
... handle = sess.add_torrent(
...     {'save_path':'', 'url':'magnet:?xt=urn:btih:5FKOMYDL555S6I6YNFKCSEWGOWC3\n'})
... print(str(handle.info_hash()), handle.is_valid())
... handle = sess.add_torrent(
...     {'save_path':'', 'uri':'magnet:?xt=urn:btih:5FKOMYDL555S6I6YNFKCSEWGOWC3\n'})
... print(str(handle.info_hash()), handle.is_valid())
('0000000000000000000000000000000000000000', True)
('0000000000000000000000000000000000000000', True)
>>> 
==================== New Session ====================
>>> import libtorrent as lt
... 
... sess = lt.session()
... handle = sess.add_torrent(
...     {'save_path':'', 'url':'magnet:?xt=urn:btih:5FKOMYDL555S6I6YNFKCSEWGOWC3'})
... print(str(handle.info_hash()), handle.is_valid())
... handle = sess.add_torrent(
...     {'save_path':'', 'uri':'magnet:?xt=urn:btih:5FKOMYDL555S6I6YNFKCSEWGOWC3'})
... print(str(handle.info_hash()), handle.is_valid())
('e954e6606bef7b2f23d869542912c67585000000', True)
('0000000000000000000000000000000000000000', True)
```

**Comment by arvidn** [comment]
testing this in RC_1_1, add_torrent() throws an exception (saying ``invalid_info_hash``). I will add that test to make sure it's the case in master as well.

**Comment by cas--** [comment]
So realised I had actually downgraded to 1.0.11 before testing those examples but if RC_1_1 throws that error then that is worse because it's then completely broken when supplying a magnet url.

**Comment by arvidn** [comment]
can you explain why throwing on invalid input is bad?

**Comment by cas--** [comment]
Because if you have the url you shouldn't need the info hash

**Comment by arvidn** [comment]
are you saying that if you have an http url to download a .torrent file from, you don't need the info-hash? In that case you don't need the info hash and it won't fail (however, http functionality is deprecated).

The case I'm talking about is where you pass in a magnet link with an invalid info-hash. Then it will throw, and I don't see how that's bad.

**Comment by cas--** [comment]
Sorry I was away from the computer, I see the confusion, that was meant to be a valid magnet url with a trailing newline but at some point 4 chars have gone awol!

`5FKOMYDL555S6I6YNFKCSEWGOWC3`
instead of:
`5FKOMYDL555S6I6YNFHUSUKCSEWGOWC3`

Testing on 1.1.1 I can confirm this has been fixed so it's isolated to 1.0.11. I am happy to close this then.
